pkg/export/setup/setup.go

```diff
@@ -50,6 +50,9 @@ const (
 	UAModeUnspecified = "unspecified"
 	UAModeAVMW        = "on-prem"
 	UAModeABM         = "baremetal"
+	UAModeAWS         = "aws"
```

Comment:
We have both a `UserAgentMode` and `UserAgentEnv`, where the former is generally reserved for the mode of installation and the latter for the platform environment. Notably, the former is user-modifiable via a flag, whereas the latter is hard-coded into the binary and auto-populated at runtime.

Is there any way to auto-detect whether we're running in AWS or Azure using a go library that we could add to the logic in [`setup.go`](https://github.com/GoogleCloudPlatform/prometheus-engine/blob/4e7649c1e50dd7ea326d2760b954258a41f53349/pkg/export/setup/setup.go#L93-L118)?

For on-prem or baremetal, autodetection is not possible due to the heterogeneity of those environments, so using "mode" as a way to distinguish these makes sense, albeit it's getting double-usage out of that flag.

Reply:
That's a good point. @ericroman920 could you comment on the auto detection for anthos on aws & azure?